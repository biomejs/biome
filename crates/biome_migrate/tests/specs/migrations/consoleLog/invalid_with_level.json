{
  "linter": {
    "rules": {
      "suspicious": {
        "noConsoleLog": {
          "level": "warn"
        }
      }
    }
  }
}
