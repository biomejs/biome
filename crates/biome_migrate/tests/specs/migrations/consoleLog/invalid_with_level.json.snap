---
source: crates/biome_migrate/tests/spec_tests.rs
expression: invalid_with_level.json
snapshot_kind: text
---
# Input
```json
{
  "linter": {
    "rules": {
      "suspicious": {
        "noConsoleLog": {
          "level": "warn"
        }
      }
    }
  }
}

```

# Diagnostics
```
invalid_with_level.json:5:9 migrate  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! The rule noConsoleLog has been removed.
  
    3 │     "rules": {
    4 │       "suspicious": {
  > 5 │         "noConsoleLog": {
      │         ^^^^^^^^^^^^^^
    6 │           "level": "warn"
    7 │         }
  
  i Safe fix: Use the rule useConsole instead.
  
     3  3 │       "rules": {
     4  4 │         "suspicious": {
     5    │ - ········"noConsoleLog":·{
     6    │ - ··········"level":·"warn"
        5 │ + ········"noConsole":·{
        6 │ + ··········"level":·"warn",
        7 │ + ··········"options":·{
        8 │ + ············"allow":·["log"]
        9 │ + ··········}
     7 10 │           }
     8 11 │         }
  

```
